### PR TITLE
ROB: Relax flate decoding for too many lookup values

### PR DESCRIPTION
**Pull request by stefan6419846**
When handling flate objects with a lookup table and the image mode `1`, we would previously raise a generic `AssertionError` if the number of lookup values did not match.

This PR proposes to add a more meaningful error message. Additionally, cases where too many values are specified are now considered a warning only as I could not see any real difference.

I might try to find a document version I can use for public test cases later on to at least cover the case where there are too many values.

### CODE REVIEW

**Comment by stefan6419846** [comment]
Example: We have the lookup table `b'\x00\x00\x00\xff\xff\xff\n'`, id est with a trailing whitespace character. This is generated by the SkyPDF Pro Driver for example.

File: pypdf/_xobj_image_helpers.py [hunk 1]
```diff
@@ -195,7 +195,14 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
...
             if img.mode == "1":
                 # Two values ("high" and "low").
-                assert len(lookup) == 2 * nb, len(lookup)
+                if len(lookup) != 2 * nb:
+                    if len(lookup) < 2 * nb:
+                        raise PdfReadError(f"Not enough lookup values: Expected {2 * nb}, got {len(lookup)}.")
+                    logger_warning(
+                        f"Expected {2 * nb} lookup values, got {len(lookup)}. Ignoring trailing ones.",
+                        __name__,
+                    )
```
Comment (pubpub-zz) [inline pypdf/_xobj_image_helpers.py:204]: I would more likely  extract the trailing characters, strip all WHITESPACES and check whether the result is an empty string: if not this means the data will be corrupted and so an exception should be raised

Comment (stefan6419846) [reply pypdf/_xobj_image_helpers.py:204]: Thanks for the hint. I will have a look at this tomorrow - the initial change had been written before I dug deeper into why the lookup is too long.

**Comment by MartinThoma** [comment]
@stefan6419846 Please correct me if I'm wrong, but I don't think it's a bugfix-PR.

A bug in pypdf would mean that pypdf is either...

* ... not acting according to the PDF specification, or
* ... not acting according to the pypdf documentation (directly or indirectly, if the user could get a reasonable expectation for a specific behavior)

This PR has two components:

* ROB: Not raising an exception but just throwing a warning in case of too many values.
* STY: Throwing a more specific error message / different types of error message

As robustness improvements (ROB) are typically more important for pypdf users, I'd start the PR with "ROB".

See https://pypdf.readthedocs.io/en/latest/dev/intro.html (I should probably extend those :-) )

**Comment by MartinThoma** [comment]
From my side this looks good to be merged :+1: Just let me know if I can change the PR title

**Comment by stefan6419846** [comment]
@MartinThoma No worries, I am completely fine with you adjusting the title if required, referrring to robustness instead. And adjusting the dev docs sounds like a nice idea as well to make it more clear.

Feel free to merge this PR after updating the title. If I find some time during the next week, I might provide some more test data for the two new edge cases to increase coverage in a separate PR - for now, I just did not stumble upon such files as in theory they violate the PDF standard anyway, while the whitespace stuff seems more spec-like and thus appears in the real world as well.
